ngi_reports/common/ign_sample_report.py

```diff
@@ -363,34 +370,39 @@ def make_plots(self):
 
 
 
-    def check_fields(self):
+    def check_project_fields(self):
         """ Check that the object has all required fields. Returns True / False.
         """
         report_fields = []
         project_fields = ['id', 'sequencing_centre', 'sequencing_platform', 'ref_genome']
-        sample_fields = ['total_reads',  'percent_aligned', 'aligned_reads', 'median_insert_size',
-            'automsomal_coverage', 'ref_above_30X', 'percent_gc']
-        plot_fields = ['coverage_plot', 'cov_frac_plot', 'insert_size_plot', 'gc_dist_plot', 'snpEFf_plot']
 
         for f in report_fields:
-            if f not in self.info.keys():
-                self.LOG.error('Mandatory report field missing: '+f)
+            if not self.info.has_key(f):
+                self.LOG.error('Mandatory report field missing: {}'.format(f))
                 return False
         for f in project_fields:
-            if f not in self.project.keys():
+            if not self.project.has_key(f):
                 import json
                 print(json.dumps(self.project, indent=4))
-                self.LOG.error('Mandatory project field missing: '+f)
+                self.LOG.error('Mandatory project field missing: {}'.format(f))
+                return False
+        return True
+
+    def check_sample_fields(self, sample_id):
+        """ Check that the object has all required fields. Returns True / False.
+        """
+        sample_fields = ['total_reads',  'percent_aligned', 'aligned_reads', 'median_insert_size',
+            'automsomal_coverage', 'ref_above_30X', 'percent_gc']
+        plot_fields = ['coverage_plot', 'cov_frac_plot', 'insert_size_plot', 'gc_dist_plot', 'snpEFf_plot']
+
+        for f in sample_fields:
+            if not self.samples[sample_id].has_key(f):
```

Comment:
**FYI**: About `has_key` (I think I have also mentioned it earlier) :wink: I am just sharing what I [read](http://stackoverflow.com/questions/1323410/has-key-or-in), so python gurus out there should correct me if it is wrong :)

```
$ python -mtimeit -s'd=dict.fromkeys(range(99))' '12 in d'
10000000 loops, best of 3: 0.0983 usec per loop
$ python -mtimeit -s'd=dict.fromkeys(range(99))' 'd.has_key(12)'
1000000 loops, best of 3: 0.21 usec per loop
```